micall/core/aln2counts.py

```diff
@@ -1358,74 +1504,123 @@ def add_nuc_read(self, offset_sequence, count):
+            insertion_coverage = {}  # max coverage left and right of the insertion
+            for left, right in insert_ranges:
+                current_insert_behind, current_insert_coverage =\
+                    get_insertion_info(right, report_aminos, report_nucleotides)
+                insert_behind[left] = current_insert_behind
+                insertion_coverage[left] = current_insert_coverage
+                current_nuc_counts = Counter()
+                insert_nuc_counts[left] = current_nuc_counts
+                for nuc_seq, count in self.nuc_seqs.items():
+                    insert_nuc_seq = nuc_seq[left:right]
+                    is_valid = (insert_nuc_seq and
+                                'n' not in insert_nuc_seq and
+                                '-' not in insert_nuc_seq)
+                    if is_valid:
+                        current_nuc_counts[insert_nuc_seq] += count
+
+            for left, counts in insert_nuc_counts.items():
+                pos = insert_behind.get(left)
+                if pos is None:
+                    # this can happen if the insertion is at the end of an alignment,
+                    # because we only look at the consensus on the right edge of the insert
+                    continue
+                ref_pos = pos - 1
+                self.ref_insertions[region][ref_pos] =\
+                    aggregate_insertions(insert_nuc_counts[left],
+                                         consensus_pos=left-1,
+                                         coverage_nuc=insertion_coverage[left])
+                insert_pos = insert_behind.get(left) // 3
+                count = sum(counts.values())
                 # Only care about insertions in the middle of the sequence,
                 # so ignore any that come before or after the reference.
                 # Also report if we're in test mode (no report_aminos).
-                if not report_aminos or insert_before not in (1, None):
-                    row = dict(seed=self.seed,
-                               region=region,
-                               qcut=self.qcut,
-                               left=left + 1,
-                               insert=insert_seq,
-                               count=count,
-                               before=insert_before)
-                    self.insert_writer.writerow(row)
-                    if insert_before is not None:
-                        region_insert_pos_counts[insert_before-1] += count
+                if not report_aminos or insert_pos not in (0, None):
+                    if insert_pos is not None:
+                        region_insert_pos_counts[insert_pos] += count
+
+        self.parse_conseq_insertions(seed_name, report_nucleotides_all)
+        self.write_insertions_file(landmarks, consensus_builder)
+
+    def parse_conseq_insertions(self, seed_name, report_nucleotides_all):
+        for insertion_position in self.conseq_insertions[seed_name].keys():
```

Comment:
A dictionary's default iterator is the same as `.keys()`. You can use `for x in my_dict:`.

Reply:
Removed that, and a couple more `.keys()` that were not needed